File: packages/java-sdk/src/main/java/dev/blindfold/sdk/models/BatchResponse.java
package dev.blindfold.sdk.models;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class BatchResponse<T> {
    private List<T> results;
    private int total;
    private int succeeded;
    private int failed;

    public BatchResponse() {}

    public List<T> getResults() { return results; }
    public int getTotal() { return total; }
    public int getSucceeded() { return succeeded; }
    public int getFailed() { return failed; }
}

File: packages/dotnet-sdk/src/Blindfold.Sdk/Models/BatchResponse.cs
using System.Text.Json.Serialization;

namespace Blindfold.Sdk.Models;

public class BatchResponse<T>
{
    [JsonPropertyName("results")]
    public List<T> Results { get; set; } = new();

    [JsonPropertyName("total")]
    public int Total { get; set; }

    [JsonPropertyName("succeeded")]
    public int Succeeded { get; set; }

    [JsonPropertyName("failed")]
    public int Failed { get; set; }
}
